File: 1538-MaximumPointsYouCanObtainFromCards/1538-MaximumPointsYouCanObtainFromCards.py
# Last updated: 5/29/2026, 11:53:18 AM
class Solution:
    def maxScore(self, cardPoints: List[int], k: int) -> int:
        wsum=0
        for i in range(k):
            wsum+=cardPoints[i]
        max_val=wsum
        for i in range(k):
            wsum=wsum-cardPoints[k-i-1]+cardPoints[len(cardPoints)-i-1]
            max_val=max(max_val,wsum)
        return max_val

File: 1538-MaximumPointsYouCanObtainFromCards/1538-MaximumPointsYouCanObtainFromCards.java
// Last updated: 5/29/2026, 11:53:24 AM
class Solution {
    public int maxScore(int[] cardPoints, int k) {
        int wsum=0;
        for(int i=0;i<k;i++){
            wsum=wsum+cardPoints[i];
        }
        int max=wsum;
        for(int i=0;i<k;i++){
            wsum=wsum-cardPoints[k-i-1]+cardPoints[cardPoints.length-i-1];
            max=Math.max(max,wsum);
        }
        return max;
        
    }
}